### PR TITLE
Added DriverComposite and implementation.

### PR DESCRIPTION
**Pull request by WeronikaMarciniak**
Adjusted tests to work with DriverComposite.

### CODE REVIEW

File: src/main/java/edu/kis/powp/jobs2d/drivers/composite/IDriverComposite.java [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+public interface IDriverComposite extends Job2dDriver {
+
+
+    @Override
+    void setPosition(int x, int y);
+    @Override
+    void operateTo(int x, int y);
+    void add(Job2dDriver driver);
+    void add(LineDriverAdapter driver);
```
Comment (kowallus) [inline src/main/java/edu/kis/powp/jobs2d/drivers/composite/IDriverComposite.java]: why separate add? 
remove could be useful.

File: src/main/java/edu/kis/powp/jobs2d/drivers/composite/DriverComposite.java [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    public DriverComposite(){
+        this.drivers = new ArrayList<Job2dDriver>();
+    }
+    public DriverComposite(Job2dDriver driver){
+        this.drivers = new ArrayList<Job2dDriver>();
+        this.drivers.add(driver);
+    }
+    @Override
+    public void setPosition(int x, int y) {
+        this.drivers.get(this.drivers.size()-1).setPosition(x,y);
```
Comment (kowallus) [inline src/main/java/edu/kis/powp/jobs2d/drivers/composite/DriverComposite.java]: I the composite approach, the command should be propagated to all children.

File: src/main/java/edu/kis/powp/jobs2d/drivers/composite/DriverComposite.java [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+    @Override
+    public void add(Job2dDriver driver) {
+        this.drivers.add(driver);
+    }
+
+    @Override
+    public void add(LineDriverAdapter driver) {
+        this.drivers.add(driver);
+    }
```
Comment (kowallus) [inline src/main/java/edu/kis/powp/jobs2d/drivers/composite/DriverComposite.java]: What is this for?

File: src/test/java/edu/kis/powp/jobs2d/TestJobs2dApp.java [hunk 4]
```diff
@@ -55,16 +56,17 @@ private static void setupCommandTests(Application application) {
...
+		DriverComposite driverComposite = new DriverComposite();
+		driverComposite.add(new LoggerDriver());
+		DriverFeature.addDriver("Logger driver", driverComposite.getDriver(0));
 
 		DrawPanelController drawerController = DrawerFeature.getDrawerController();
-		Job2dDriver driver = new LineDriverAdapter(drawerController, LineFactory.getBasicLine(), "basic");
-		DriverFeature.addDriver("Line Simulator", driver);
-		DriverFeature.getDriverManager().setCurrentDriver(driver);
+		driverComposite.add(new LineDriverAdapter(drawerController, LineFactory.getBasicLine(), "basic"));
+		DriverFeature.addDriver("Line Simulator", driverComposite.getDriver(1));
```
Comment (kowallus) [inline src/test/java/edu/kis/powp/jobs2d/TestJobs2dApp.java]: This doesn't make sense to me.

File: src/main/java/edu/kis/powp/jobs2d/drivers/composite/IDriverComposite.java [hunk 5]
```diff
@@ -0,0 +1,17 @@
+package edu.kis.powp.jobs2d.drivers.composite;
+import edu.kis.powp.jobs2d.Job2dDriver;
+import edu.kis.powp.jobs2d.drivers.adapter.LineDriverAdapter;
+
+
+public interface IDriverComposite extends Job2dDriver {
+
+
+    @Override
```
Comment (kowallus) [inline src/main/java/edu/kis/powp/jobs2d/drivers/composite/IDriverComposite.java:8]: Wrong formatting convention. Project uses tabs (not spaces).

**Review by kowallus — CHANGES_REQUESTED** [review]
I have some suggestions. 
Moreover, instead of modifying already existing drivers (in the app), add a new one which will be a composite of logger and drawing drivers (two in one).

File: src/main/java/edu/kis/powp/jobs2d/drivers/composite/IDriverComposite.java (hunk 1 above)
Comment (Jakubszadkowski) [reply src/main/java/edu/kis/powp/jobs2d/drivers/composite/IDriverComposite.java]: Remove added. Separate add was removed.

File: src/main/java/edu/kis/powp/jobs2d/drivers/composite/DriverComposite.java (hunk 2 above)
Comment (Jakubszadkowski) [reply src/main/java/edu/kis/powp/jobs2d/drivers/composite/DriverComposite.java]: Adjusted that. Should be good now.

File: src/main/java/edu/kis/powp/jobs2d/drivers/composite/DriverComposite.java (hunk 3 above)
Comment (Jakubszadkowski) [reply src/main/java/edu/kis/powp/jobs2d/drivers/composite/DriverComposite.java]: Removed.

File: src/main/java/edu/kis/powp/jobs2d/drivers/composite/IDriverComposite.java (hunk 5 above)
Comment (Jakubszadkowski) [reply src/main/java/edu/kis/powp/jobs2d/drivers/composite/IDriverComposite.java:8]: Switched from spaces to tabs.

File: src/test/java/edu/kis/powp/jobs2d/TestJobs2dApp.java (hunk 4 above)
Comment (Jakubszadkowski) [reply src/test/java/edu/kis/powp/jobs2d/TestJobs2dApp.java]: Just using composite's list to use drivers.

File: src/test/java/edu/kis/powp/jobs2d/TestJobs2dApp.java [hunk 6]
```diff
@@ -69,6 +69,10 @@ private static void setupDrivers(Application application) {
 
 		driverComposite.add(new LineDriverAdapter(drawerController, LineFactory.getSpecialLine(), "special"));
 		DriverFeature.addDriver("Special line Simulator", driverComposite.getDriver(2));
+
+		DriverFeature.addDriver("Driver composite", driverComposite);
```
Comment (kowallus) [inline src/test/java/edu/kis/powp/jobs2d/TestJobs2dApp.java:77]: OK, but revert changes to previous addDriver calls (to make code more clear).

Comment (Jakubszadkowski) [reply src/test/java/edu/kis/powp/jobs2d/TestJobs2dApp.java:77]: Reverted.

**Pull request merged**